libs/entitlement/bridge/src/main/java/org/elasticsearch/entitlement/bridge/EntitlementChecker.java

```diff
@@ -254,4 +258,37 @@ public interface EntitlementChecker {
     void check$java_net_Socket$connect(Class<?> callerClass, Socket that, SocketAddress endpoint);
 
     void check$java_net_Socket$connect(Class<?> callerClass, Socket that, SocketAddress endpoint, int backlog);
+
+    // Network miscellanea
+    void check$java_net_URL$openConnection(Class<?> callerClass, java.net.URL that, Proxy proxy);
+
+    // HttpClient.Builder is an interface, so we instrument its only (internal) implementation
+    void check$jdk_internal_net_http_HttpClientBuilderImpl$build(Class<?> callerClass, HttpClient.Builder that);
+
+    // HttpClient#send and sendAsync are abstract, so we instrument their internal implementation
```

Comment:
These will silently stop providing protection if ES is run on any JDK that doesn't use this same implementation class.

But this is a known problem we need to wrestle with, and it doesn't need to stand in the way of merging this PR.